draft-ietf-mls-protocol.md

```diff
@@ -3607,23 +3612,12 @@ uses:
 
 ### Creating a Commit
 
-When creating or processing a Commit, three different ratchet trees and
-their associated GroupContexts are used:
-
-1. "Old" refers to the ratchet tree and GroupContext for the epoch before the
-   commit.  The old GroupContext is used when signing the MLSPlainText so that
-   existing group members can verify the signature before processing the
-   commit.
-2. "Provisional" refers to the ratchet tree and GroupContext constructed after
-   applying the proposals that are referenced by the Commit.  The provisional
-   GroupContext uses the epoch number for the new epoch, and the old confirmed
-   transcript hash.  This is used when creating the UpdatePath, if the
-   UpdatePath is needed.
-3. "New" refers to the ratchet tree and GroupContext constructed after applying
-   the proposals and the UpdatePath (if any).  The new GroupContext uses the
-   epoch number for the new epoch, and the new confirmed transcript hash.  This
-   is used when deriving the new epoch secrets, and is the only GroupContext
-   that newly-added members will have.
+When creating or processing a Commit, a client updates the ratchet tree and
+GroupContext for the group.  These values advance from an "old" state reflecting
+the current epoch to a "new" state reflecting the new epoch initiated by the
+Commit.  When the Commit includes an UpdatePath, a "provisional" group context
+is constructed that reflects changes due to the proposals and UpdatePath, but
+with the old confirmed transcript hash.
```

Comment:
If you want to keep the mental models, you could add:
"Conceptually, these three states can be thought of as three separate ratchet trees."